decavision/model_testing/testing.py

```diff
@@ -253,3 +256,198 @@ def generate_classification_report(self, path):
+    
+    def __init__(self, model, json_file, categories):
+        use_tpu, use_gpu = utils.check_PU()
+        if use_tpu:
+            # necessary because keras generators don'T work with TPUs...
+            tf.compat.v1.disable_eager_execution()
+        try:
+            self.model = load_model(model, custom_objects={"_f1_score": f1_score, "f1_score": f1_score})
+            # efficientnets have the scaling included in them so no need to rescale the images when loading
+            if self.model.name[0] in ['B', 'V']:
+                self.rescaling = 1
+            else:
+                self.rescaling = 255
+            print('Model loaded correctly')
+        except Exception as e:
+            print('There was a problem when trying to load your model: {}'.format(e))
+
+        data = []
+        with open(json_file, 'r') as file:
+            values = json.load(file)
+            for f, l in values.items():
+                data.append({"filenames": f, "labels": l})
+
+        self.input_shape = self.model.input_shape[1:3]
+        self.df = pd.DataFrame(data)
+        self.values = values
+        self.json_file = json_file
+        self.categories = categories
+        print("Model name: ",self.model.name)
+
+
+    def _load_dataset(self, path):
+        """
+        Load dataset into a keras generator. Images must be contained in single
+        folder for each class. A dataframe is required with 2 columns: original 
+        image name and image labels as a list separated by comma. i.e.
+        
+        filenames                |    labels
+        -------------------------|-----------------
+        sun_abifwwwgjnomvfda.jpg | [asphalt,clouds,natural_light,man-made,open_area,far-away_horizon,sky,barn,airfield]
+        sun_aciuvbhvntkgdhhk.jpg | [grass,asphalt,natural_light,natural,man-made,open_area,far-away_horizon,sky,dirt/soil,airfield]
+
+        Arguments:
+            path (str): location of the dataset
+
+        Returns:
+            generator: images plus information about them (labels, paths, etc)
+        """
+                   
+        datagen = tf.keras.preprocessing.image.ImageDataGenerator(rescale=1 / self.rescaling)
+        generator = datagen.flow_from_dataframe(directory=path,
+                                                dataframe=self.df,
+                                                x_col="filenames",
+                                                y_col="labels",
+                                                target_size=self.input_shape,
+                                                shuffle=False,
+                                                color_mode='rgb',
+                                                class_mode='categorical',
+                                                classes=self.categories,
+                                                batch_size=1)    
+        return generator
+
+    def classify_images(self, path, categories, threshold = 0.5, plot=True):
```

Comment:
You can use `self.categories` instead of asking them as input.

Reply:
done.